src/hotkey/key_window.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <stdint.h>
+
+/*
+ * The key_window module provides a FIFO key_window interface to help with bookkeeping the
```

Comment:
Given that we don't really update the size of key window, does it make things easier to use ring array to implement this data structure?